### DISCUSSION TITLE
Typehinting function return with comment-type hints

### DISCUSSION THREAD

**Discussion by Kein** [microsoft/pylance-release#5545] [Q&A]
I remember about 2 or so years ago it used to work, I could typehint the return with something like:
![image](https://github.com/microsoft/pylance-release/assets/623756/3dd73eec-c718-4253-a1ca-ae1b44b8d00e)

But as you can see it no longer works. Is there any way to make it work again? An option or a fork?

**Comment by erictraut** [comment]
A simple `# type: X` hint would have never worked. However, there are historical (Python 2-era) type hint forms that can be used. You can read about them [here](https://typing.readthedocs.io/en/latest/spec/historical.html#type-comments-on-function-definitions). However, I strongly recommend against using these deprecated forms, as support for them will likely go away in the near future. Most modern tooling for Python has already removed support for them. The correct way to add type annotations in Python 3.x code is to use the modern syntax (` -> str | None`).

**Reply by Kein** [reply]
Thank you, eric. Yeah, I know it is outdated but I need it for compatibility reasons.
